### PR TITLE
Improvements

### PR DESCRIPTION
**Pull request by Jannis**
@fordN I decided to submit a PR against your PR instead of a review. This addresses all the comments I would've had:

* Shorter / more simple warning messages.
* Use the built-in `spinner.warn` instead of a custom `stopAndPersist` call.
* Improve `warning` object check in `withSpinner`.
* Add empty line after warnings to balance things visually.

Let me know what you think!

Screenshot:

image

### CODE REVIEW

**Comment by Jannis** [comment]
Hmm, ok, I've decided to bring the `from` back in the warning caption:

image

**Comment by Jannis** [comment]
I'll fix the tests tomorrow.

**Comment by Jannis** [comment]
Tests are fixed — but Node 11 seems to fail on Travis (even on master). Let's ignore that for the moment and look into it when we have time.

File: src/command-helpers/spinner.js [hunk 1]
```diff
@@ -14,19 +14,27 @@ const step = (spinner, subject, text) => {
...
   try {
     let result = await f(spinner)
-    if(warningText && immutable.Map.isMap(result)) {
-      if (result.get('warning', false)) {
-        spinner.stopAndPersist({symbol: '⚠️', text: `${warningText}: ${result.get('warning')}`})
-        return result.get('result')
+    if (typeof result === 'object') {
+      if (result.warning && Object.keys(result).indexOf('result') >= 0) {
+        spinner.warn(`${warningText}: ${result.warning}`)
+        spinner.succeed(text)
```
Comment (fordN) [inline src/command-helpers/spinner.js]: Is it necessary to have both `spinner.warn` and `spinner.succeed` when there is a warning?

**Review by fordN — CHANGES_REQUESTED** [review]

File: src/command-helpers/spinner.js (hunk 1 above)
Comment (Jannis) [reply src/command-helpers/spinner.js]: Yep, because we want to see `[x] Load subgraph` even if there were warnings.

**Comment by Jannis** [comment]
@fordN I feel I've addressed your point. Anything else? 😉

File: src/command-helpers/spinner.js (hunk 1 above)
Comment (fordN) [reply src/command-helpers/spinner.js]: My understanding is that the loading text (in this case "Load subgraph from...") is set by the first line, `let spinner = toolbox.print.spin(text)`.   I don't see any difference in the output with the line in question removed.

Comment (Jannis) [reply src/command-helpers/spinner.js]: First with the extra `succeed`, then without it:
image

Only in the first example is `✔ Load subgraph from subgraph.yaml` logged after the warnings. And that's what we want.

**Comment by fordN** [comment]
👍

**Review by fordN — APPROVED** [review]

**Pull request merged**